web/interceptor/access_control.go

```diff
@@ -43,6 +43,7 @@ var accessRolesFor = map[string]roles{
 	"UpdateUser":              {user, admin},
 	"GetEnrollmentsByUser":    {user, admin},
 	"GetSubmissions":          {student, group, teacher},
+	"GetSubmission":           {teacher},
```

Comment:
Should students be allowed to invoke `GetSubmission`? Or do they only need `GetSubmissions`?

Reply:
We could allow students to invoke `GetSubmission`. This will require checking that the user ID of the submission matches the user ID in the incoming context. If the submission is for a group assignment, we need to match the user ID with the members of the group.

If we do allow them to invoke the method, we could add a `Refresh` button in the frontend (or some similar functionality) to refresh the submission they are currently viewing.